src/main/java/generics/Box.java

```diff
@@ -0,0 +1,40 @@
+package generics;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Box<T extends Fruit> {
+    private final List<T> fruitList;
```

Comment:
```suggestion
    private final List<T> fruits;
```
Немає сенсу наголошувати, що це список. Змінювати не треба, для себе просто запам'ятай - краще називати в множині, ніж додавати List, Set, Map і так далі.